labop/execution_context.py

```diff
@@ -0,0 +1,376 @@
+from typing import Dict, Iterable, List, Optional
+
+import sbol3
+
+from labop import parameter_value
+from labop.call_behavior_execution import CallBehaviorExecution
+from uml import (
+    Action,
+    Activity,
+    ActivityEdge,
+    ActivityNode,
+    ActivityParameterNode,
+    CallBehaviorAction,
+    ControlFlow,
+    LiteralSpecification,
+    ObjectFlow,
+    OutputPin,
+    Pin,
+    ValuePin,
+    literal,
+)
+from uml.activity import Activity
+from uml.activity_edge import ActivityEdge
+from uml.behavior import Behavior
+from uml.final_node import FinalNode
+from uml.initial_node import InitialNode
+
+from .parameter_value import ParameterValue
+
+
+class ExecutionContext(object):
```

Comment:
This is apparently a utility class for the software, and not yet part of the LabOP data model.  Should it be?

(Also, at this point in the review, I haven't quite grocked how it is used, but maybe that will come clear later...)

Reply:
This is actually a pretty important addition to the execution engine.  It is essentially a stack frame, but because we don't have a stack, its called a context. Every protocol will have at least two contexts. The outer context invokes a call behavior action that corresponds to the protocol.  This simplifies how the EE handles tokens because it associates pins with the protocol inputs and outputs.  It also helps to manage execution when there are many nested protocols.